llama-index-core/llama_index/core/vector_stores/simple.py

```diff
@@ -224,6 +241,37 @@ def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
             if self._data.metadata_dict is not None:
                 self._data.metadata_dict.pop(text_id, None)
 
+    def delete_nodes(
+        self,
+        node_ids: Optional[List[str]] = None,
+        filters: Optional[MetadataFilters] = None,
+        **delete_kwargs: Any,
+    ) -> None:
+        filter_fn = _build_metadata_filter_fn(
+            lambda node_id: self._data.metadata_dict[node_id], filters
+        )
```

Comment:
Great job!

Maybe it's better to use `self._data.metadata_dict.get(node_id, {})` to avoid KeyError?

Reply:
Oh, never mind, I just found node_id is from `self._data.embedding_dict` on line 265.